File: core/src/main/java/org/selyu/smp/core/item/impl/HellstoneIngotItem.java
package org.selyu.smp.core.item.impl;

import org.bukkit.Material;
import org.jetbrains.annotations.NotNull;
import org.selyu.smp.core.item.CustomItem;
import org.selyu.smp.core.item.CustomItemType;
import org.selyu.smp.core.item.recipe.Recipe;
import org.selyu.smp.core.item.recipe.ShapedRecipeBuilder;

public final class HellstoneIngotItem extends CustomItem {
    public HellstoneIngotItem() {
        super(CustomItemType.HELLSTONE_INGOT, Material.PAPER, 1);
    }

    @NotNull
    @Override
    public String getDisplayName() {
        return "&fHellstone Ingot";
    }

    @Override
    public @NotNull Recipe getRecipe() {
        return new ShapedRecipeBuilder(this)
                .shape(
                        'm', 'm', 'm',
                        'm', 'i', 'm',
                        'm', 'm', 'm'
                )
                .ingredient('m', Material.MAGMA_BLOCK)
                .ingredient('i', Material.IRON_INGOT)
                .build();
    }
}

File: core/src/main/kotlin/org/selyu/smp/core/item/impl/HellstoneIngotItem.kt
package org.selyu.smp.core.item.impl

import org.bukkit.Material
import org.selyu.smp.core.item.CoreItem
import org.selyu.smp.core.item.CoreItemType
import org.selyu.smp.core.item.CoreRecipe
import org.selyu.smp.core.item.MaterialRecipeKey

class HellstoneIngotItem : CoreItem("hellstone_ingot", CoreItemType.INGOT, Material.PAPER, 1) {
    override fun getDisplayName(): String = "Hellstone Ingot"
    override fun getLore(): MutableList<String> = mutableListOf("&c&oForged from fiery depths...")

    override fun getRecipe(): CoreRecipe? = CoreRecipe(this).also {
        it.addKey(MaterialRecipeKey(Material.MAGMA_BLOCK, 8))
        it.addKey(MaterialRecipeKey(Material.IRON_INGOT))
    }
}